src/hotspot/share/compiler/compileBroker.cpp

```diff
@@ -1129,10 +1132,11 @@ void CompileBroker::possibly_add_compiler_threads(Thread* THREAD) {
     for (int i = old_c1_count; i < new_c1_count; i++) {
       JavaThread *ct = make_thread(compiler_t, compiler1_object(i), _c1_compile_queue, _compilers[0], THREAD);
       if (ct == NULL) break;
+      ThreadsListHandle tlh;  // get_thread_name() depends on the TLH.
+      assert(tlh.includes(ct), "ct=" INTPTR_FORMAT " exited unexpectedly.", p2i(ct));
```

Comment:
This can be placed inside the if-clause where the MutexLocker was.

Reply:
True. That would be better.